src/cli.py

```diff
@@ -1,2 +1,17 @@
 class Cli():
-    pass
+
+    def log(self, message, writer=print):
```

Comment:
I like how you have print as the default writer. Makes this easy to inject a mock.

Reply:
Yeah, after much deliberation with Ian, videos, and blog posts this is what I came up with to handle it